.github/workflows/e2e-tests.yml

```diff
@@ -12,20 +12,87 @@ on:
         type: string
 
 jobs:
+
   e2e:
-    name: "E2E Tests"
+    name: E2E Tests - Shard ${{ matrix.shard }} of ${{ matrix.total_shards }}
+    runs-on: ubuntu-latest
+
+    strategy:
+        matrix:
+          shard: [1, 2, 3]
+          total_shards: [3]
```

Comment:
Add a comment for this line that explains that GitHub Actions doesn't have a built-in method to get the length of an array

Reply:
I think you can technically push the length - but it's probably not worth it to add a step

```
      - name: Calculate total shards
        id: shard_length
        run: echo "TOTAL_SHARDS=${{ matrix.shard | length }}" >> $GITHUB_ENV

```